### PR TITLE
Generic gyros for gyro failover

### PR DESCRIPTION
**Pull request by having11**
Automatically track failed gyros and move to the next working one seamlessly

### CODE REVIEW

File: src/main/cpp/subsystems/DriveSubsystem.cpp [hunk 1]
```diff
@@ -261,3 +243,22 @@ void DriveSubsystem::ResetOdometry(frc::Pose2d pose) {
...
+          "DriveSubsystem", "Gyro %s has had error '%s'.", gyro->GetName(),
+          gyro->GetStatusMessage());
+      m_lastGoodGyroIndex++;
+      continue;
+    }
+    return gyro;
+  }
+
+  ConsoleLogger::getInstance().logError(
+      "DriveSubsystem", "All gyros have failed. HELP US EVAN!!!!1!!1!%s", "");
```
Comment (nobody5050) [inline src/main/cpp/subsystems/DriveSubsystem.cpp:261]: Is this the 3rd log in our code that explicitly calls out evan?

File: src/main/include/subsystems/DriveSubsystem.h [hunk 2]
```diff
@@ -165,9 +168,19 @@ class DriveSubsystem : public frc2::SubsystemBase {
...
+  PigeonGyroContainer m_gyroContainer1{m_gyro1, "First Pigeon"};
+  PigeonGyroContainer m_gyroContainer2{m_gyro2, "Second Pigeon"};
+  Adxrs450Gyro m_gyroContainer3{m_gyro3, "ADXRS450 Gyro"};
+
+  std::vector<IGyroContainer*> m_gyros = {
+      dynamic_cast<IGyroContainer*>(&m_gyroContainer1),
+      dynamic_cast<IGyroContainer*>(&m_gyroContainer2),
+      dynamic_cast<IGyroContainer*>(&m_gyroContainer3)};
+
+  // AHRS m_gyro{frc::SPI::Port::kMXP};
```
Comment (nobody5050) [inline src/main/include/subsystems/DriveSubsystem.h:182]: Begone NavX!

**Review by nobody5050 — APPROVED** [review]

File: src/main/cpp/subsystems/DriveSubsystem.cpp (hunk 1 above)
Comment (having11) [reply src/main/cpp/subsystems/DriveSubsystem.cpp:261]: @WowMuchDoge

**Comment by nobody5050** [comment]
Can probably be closed now that we’ve implemented a fix for baby auto

**Pull request closed without merging**